src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -0,0 +1,275 @@
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+
+import asyncio
+from typing import Dict, List
+
+from mdns_discovery.exceptions import DiscoveryNotPerformedError
+from zeroconf import IPVersion, ServiceStateChange, Zeroconf
+from zeroconf.asyncio import AsyncServiceBrowser, AsyncServiceInfo
+
+from .mdns_type_enum import MdnsType
+
+
+class MdnsDiscovery:
+
+    DEFAULT_DISCOVERY_DURATION_SEC = 3
+
+    def __init__(self, tc):
+        """
+        Initializes the MdnsDiscovery instance with necessary configurations.
+
+        Args:
+            tc: A test case object containing node and fabric information.
+
+        Attributes:
+            _zc (Zeroconf): An instance of Zeroconf to manage mDNS operations.
+            _aiobrowser: An instance of the AsyncServiceBrowser browser to discover services.
+            _tc: The test case object.
+            _service_types (list): A list of mDNS service types to discover.
+            _discovered_services (dict): A dictionary to store discovered services.
+            _discovery_performed (bool): Flag to indicate if discovery has been performed.
+            dut_operational_service_name (str): The operational service name for the DUT (Device Under Test).
+
+        Main methods, return a list of service info dicts, each dict has keys for querying values
+            - get_commissioner_service_info()
+            - get_commissionable_service_info()
+            - get_operational_service_info()
+            - get_border_router_service_info()
+
+        Keys:
+            - service_name: The unique name of the mDNS service.
+            - name: The human-readable name of the service.
+            - type: The type of the service, typically indicating the service protocol and domain.
+            - server: The domain name of the machine hosting the service.
+            - port: The network port on which the service is available.
+            - addresses: A list of IP addresses associated with the service.
+            - txt_record: A dictionary of key-value pairs representing the service's metadata.
+            - priority: The priority of the service, used in service selection when multiple instances are available.
+            - interface_index: The network interface index on which the service is advertised.
+            - weight: The relative weight for records with the same priority, used in load balancing.
+            - host_ttl: The time-to-live value for the host name in the DNS record.
+            - other_ttl: The time-to-live value for other records associated with the service.
```

Comment:
See my other comment about dataclass. Document those in the dataclass, field by field.